### DISCUSSION TITLE
ZXingReader can't recognize QR code in a MJPEG frame

### DISCUSSION THREAD

**Discussion by siiky** [zxing-cpp/zxing-cpp#620] [Q&A] [answered]
I have a new camera whose frames ZXing can't decode, and I can't figure out why. I've tried different V4L2 settings for colorspace and YCbCr encoding and nothing seems to make a difference. I tried the [WASM demo](https://zxing-cpp.github.io/zxing-cpp/demo_reader.html) and it's the same, which maybe excludes something wrong with/specific to my pipeline (V4L2 -> `stb_image` -> ZXing). It's not a problem with the QR codes themselves, because ZXing can decode them when using other cameras. This one is the first I'm running into problems with (maybe something wrong with this camera?).

Something weird I've found: if I convert the image with Image Magick's `convert`, or send the image through some chat and download again, then the WASM demo is able to scan the QR code. I tried comparing the two but can't find anything relevant (admittedly, I'm a fish out of water here).

Sharing a ZIP file to be sure GitHub won't mess with the original image: [qr.zip](https://github.com/zxing-cpp/zxing-cpp/files/12686176/qr.zip) (SHA512: `b1c70a682e605b552a4bb1b785c22ff89f8533ab23fd7f59b826ed424796b71ccd7489579f53045b19711e3f7b08a45029f3a62cb93d72c24cb654492d3a6d74`).

Inside there's `original.jpg` (taken straight from the camera), and `converted.jpg` (the result of `convert original.jpg converted.jpg`).

**Comment by axxel** [comment]
The problem is how `stb_image` handles an apparently unexpected marker in the JPEG byte stream. Version 2.27 of `stb_image.h` fails to open the image, version 2.28 opens it just fine but returns a white image (or so it seems). What version do you have/use?

**Reply by siiky** [reply]
I'm using v2.28, the latest as of now.

What is this "marker" you mention, and how can I learn more about it and find it? As a workaround, would it be possible to simply remove it from the image before decoding?

**Reply by siiky** [reply]
Thanks for the reply btw! :) even though this isn't a problem with ZXing.

**Reply by siiky** [reply]
Is this issue related? https://github.com/nothings/stb/issues/1409

**Reply by axxel** [reply]
I don't think so, but almost. See my report at https://github.com/nothings/stb/issues/1518.

**Reply by axxel** [reply, marked as answer]
Update: it turned out that this image is a MJPEG frame and `stb_image` does not support reading those. The https://github.com/nothings/stb/issues/1518 does contain a patch to support MJPEG files.
